lang/src/org/partiql/lang/eval/time/Time.kt

```diff
@@ -32,11 +46,11 @@ import kotlin.math.min
  *  Note that the [LocalTime] always stores the fractional part of the second in nanoseconds.
  *  It is up to the application developers to make use of the preserved [precision] value as need be.
  */
-data class Time(val localTime: LocalTime, val zoneOffset: ZoneOffset? = null, val precision: Int = MAX_PRECISION_FOR_TIME) {
+data class Time(val localTime: LocalTime, val precision: Int = MAX_PRECISION_FOR_TIME, val zoneOffset: ZoneOffset? = null) {
```

Comment:
I don't think this `MAX_PRECISION_FOR_TIME` default is necessary or desirable.

Reply:
So here's the pickle. I think we should not expose this constructor (make `private`) at all because it does not make anything of `precision` and the caller might get confused. 
Static method `Time.of` takes `precision` and rounds the fractional part of the `second`. The constructor cannot embed this logic as `localTime` is immutable.  
Since it was still exposed, the `precision`  default was chosen to be `MAX_PRECISION_FOR_TIME`. 
We probably can introduce another overload of `Time.of(localTime: LocalTime, precision: Int, timeZone: TimeZone)`

Reply:
If that is the case then this constructor should be at least `internal`, but even still the default makes no sense because internal code will never use that default, no?

Reply:
Sounds good. I will remove the default parameter and make the constructor `internal`.

Reply:
I have made the constructor `private` as it wasn't used anywhere and is not intended to be used (as is) anywhere.
I have instead added a static API `Time.of(localTime: LocalTime, precision: Int, zoneOffset: ZoneOffset? = null)` which  replaces the constructor and also uses the `precision` to round the `localTime`. I believe it is better this way.